R/query.R

```diff
@@ -199,17 +199,19 @@ convert_result_types <- function(df, coltypes_df, .use_integer64)
                 as.integer(column),
             datetime=, DateTime=
                 as.POSIXct(strptime(column, format='%Y-%m-%dT%H:%M:%OSZ', tz='UTC')),
-            real=, Double=, Float=
+            real=, double=, float=, Double=, Float=
                 as.numeric(column),
             bool=, Boolean=
-                as.logical(column),
+                       as.logical(column),
+            dynamic=
+                lapply(column, function(x) if (!is.na(x)) tryCatch(jsonlite::fromJSON(x), error=function(e) return (x))),
             as.character(column)
         )
     }
 
     df <- as.data.frame(df, stringsAsFactors=FALSE)
     names(df) <- coltypes_df$ColumnName
-    df[] <- Map(convert_kusto_datatype, df, coltypes_df$DataType, MoreArgs=list(.use_integer64=.use_integer64))
+    df[] <- Map(convert_kusto_datatype, df, coltypes_df$ColumnType, MoreArgs=list(.use_integer64=.use_integer64))
```

Comment:
This change seems to be an error, it breaks the integration tests.

Reply:
When the column is dynamic, DataType will be "string" but ColumnType will be "dynamic". I suspect it may not be handling some other data type correctly due to this change, because DataType represents the .NET nomenclature and ColumnType is the Kusto nomenclature for the types.

I will go through this list and make sure all the types we need to handle are accounted for:
https://docs.microsoft.com/en-us/azure/kusto/query/scalar-data-types/

Meanwhile can you let me know which integration test(s) failed and what the error/failure messages were, to help narrow it down?

Reply:
Ah, ok. Looks like it's only commands that are breaking--queries have been working fine for me, but ".show cluster" throws:

```
Error in mapply(FUN = f, ..., SIMPLIFY = FALSE) : 
  zero-length inputs cannot be mixed with those of non-zero length
```
Looks like I've got a little debugging to do.

Reply:
Kusto sends both "ColumnType" and "DataType" if it's a query, but only sends "DataType" if it's a command, that's why `convert_kusto_datatype()` was choking on command results. I fixed this by just copying "DataType" into "ColumnType" for command results.

Reply:
From a defensive programming POV, it would be better to check that `tables$Columns[[1]]$ColumnType` is missing before overwriting it. Also, maybe add some integration tests to test06_dplyr.R to cover the new verbs.

Reply:
Done and done!